### PR TITLE
526 multipart test

### LINKED ISSUE #526: Add multipart test to Nessie

### LINKED ISSUE #538: Gateway returned key is inconsistent with request
Key in the form of "branch/path/to/obj" is returned as "path/to/obj"

### PR DESCRIPTION
**Pull request by itaiad200**
closes #526
closes #538

### CODE REVIEW

File: gateway/operations/postobject.go [hunk 1]
```diff
@@ -54,7 +55,7 @@ func (controller *PostObject) HandleCreateMultipartUpload(o *PathOperation) {
 	}
 	o.EncodeResponse(&serde.InitiateMultipartUploadResult{
 		Bucket:   o.Repository.Name,
-		Key:      o.Path,
+		Key:      path.WithRef(o.Path, o.Reference),
```
Comment (arielshaqed) [inline gateway/operations/postobject.go:58]: I don't understand.  How did multipart uploads work without this?

File: nessie/multipart_test.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	var uploadErr error
+
+	wg.Add(numberOfParts)
+	for partNumber := 1; partNumber <= numberOfParts; partNumber++ {
+		go func(partNumber int) {
+			completedPart, err := uploadPart(logger, svc, resp, randstr.Bytes(partSize), partNumber)
+
+			if err != nil {
+				// keep just the first error
+				anyError.Do(func() {
```
Comment (arielshaqed) [inline nessie/multipart_test.go:61]: Consider just streaming out all the errors on a channel so we can report them all: when it fails, there's a big difference between one failing and multiple failures.

File: nessie/multipart_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+	logger.WithField("key", completeResponse.Key).Info("Completed multipart request successfully")
+
+	var b bytes.Buffer
+	_, err = client.GetObject(ctx, repo, masterBranch, file, &b)
+	require.NoError(t, err, "failed to get object")
+}
+
+func uploadRandomParts(t *testing.T, logger logging.Logger, resp *s3.CreateMultipartUploadOutput) []*s3.CompletedPart {
+	ch := make(chan *s3.CompletedPart, numberOfParts)
```
Comment (arielshaqed) [inline nessie/multipart_test.go:49]:
```suggestion
	t.Helper()
	ch := make(chan *s3.CompletedPart, numberOfParts)
```

**Review by arielshaqed — APPROVED** [review]
Thanks!  Approving because yes multipart uploads is better than no multipart uploads, and everything is generally good.  But see small comments.

File: gateway/operations/postobject.go (hunk 1 above)
Comment (itaiad200) [reply gateway/operations/postobject.go:58]: They work just fine if you don't pass the key that we're returning. Once you pass it back they'll fail.

**Pull request merged**